flytekit/core/context_manager.py

```diff
@@ -735,7 +735,9 @@ def new_compilation_state(self, prefix: str = "") -> CompilationState:
         Creates and returns a default compilation state. For most of the code this should be the entrypoint
         of compilation, otherwise the code should always uses - with_compilation_state
         """
-        return CompilationState(prefix=prefix)
+        from flytekit.core.python_auto_container import default_task_resolver
+
+        return CompilationState(prefix=prefix, task_resolver=default_task_resolver)
```

Comment:
why do we need this if the correct task resolver is being set in `compile` in flytekit/core/workflow.py?

Reply:
I set the task resolver in the `ctx.with_compilation_state(...)`, but there are some places where `new_compilation_state()` get called, potentially causing the loss of the resolver if it's not set.